### PR TITLE
IGNITE-16275 [Native Persistence 3.0] B+Tree porting

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-16275

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/util/IgniteRandom.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+package org.apache.ignite.internal.util;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * Random to be used from a single thread. Compatible with {@link Random} but faster.
+ */
+public class IgniteRandom extends Random {
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/util/IgniteRandom.java:26]: Is this used outside of tests?

File: modules/core/src/test/java/org/apache/ignite/internal/util/IgniteRandomSelfTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for {@link IgniteRandom}.
+ */
+public class IgniteRandomSelfTest {
```
Comment (ibessonov) [inline modules/core/src/test/java/org/apache/ignite/internal/util/IgniteRandomSelfTest.java:29]: I wonder if benchmarks could show any difference. Can we check it? I don't really trust the information that it's "faster"

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java [hunk 3]
```diff
@@ -0,0 +1,6426 @@
...
+    private final float minFill;
+
+    private final float maxFill;
+
+    /** Meta page id. */
+    protected final long metaPageId;
+
+    private boolean canGetRowFromInner;
+
+    // TODO: https://issues.apache.org/jira/browse/IGNITE-16350
```
Comment (ibessonov) [inline modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java]: I'd love to see a comment, not only a link. What's supposed to be done here?

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/IgniteTree.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+import org.apache.ignite.lang.IgniteInternalCheckedException;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Interface for ignite internal tree.
+ *
+ * @param <L> Type of keys maintained by this tree.
+ * @param <T> Type of mapped values.
+ */
+public interface IgniteTree<L, T> {
```
Comment (ibessonov) [inline modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/IgniteTree.java:30]: Please add TODO here, I'm not sure that we even need this interface

File: modules/core/src/main/java/org/apache/ignite/internal/util/IgniteRandom.java (hunk 1 above)
Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/util/IgniteRandom.java:26]: Only in tests.

Comment (tkalkirill) [reply modules/core/src/main/java/org/apache/ignite/internal/util/IgniteRandom.java:26]: Moved to tests.

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/IgniteTree.java (hunk 4 above)
Comment (tkalkirill) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/IgniteTree.java:30]: Done.

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java (hunk 3 above)
Comment (tkalkirill) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java]: Done

**Pull request merged**